backend/src/test/java/org/uresti/pozarreal/service/impl/RepresentativesServiceImplTests.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.uresti.pozarreal.exception.BadRequestDataException;
+import org.uresti.pozarreal.model.Representative;
+import org.uresti.pozarreal.repository.RepresentativeRepository;
+
+public class RepresentativesServiceImplTests {
+
+    @Test
+    public void givenRepresentativeAndUserIdEquals_whenSave_thenRepresentativeIsSaved() {
+        // Given:
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+
+        RepresentativesServiceImpl representativesService = new RepresentativesServiceImpl(representativeRepository);
+
+        Representative representative = Representative.builder()
+                .house("house")
+                .phone("123456")
+                .street("street")
+                .userId("userId")
+                .build();
+
+        org.uresti.pozarreal.dto.Representative representativeDto = org.uresti.pozarreal.dto.Representative.builder()
+                .house("house")
+                .phone("123456")
+                .street("street")
+                .userId("userId")
+                .build();
+
+        Mockito.when(representativeRepository.save(representative)).thenReturn(representative);
+
+        // When:
+        org.uresti.pozarreal.dto.Representative representativeSaved = representativesService.save(representativeDto, "userId");
+
+        // Then:
+        Assertions.assertThat(representativeSaved).isNotNull();
+        Assertions.assertThat(representativeSaved.getHouse()).isEqualTo("house");
+        Assertions.assertThat(representativeSaved.getPhone()).isEqualTo("123456");
+        Assertions.assertThat(representativeSaved.getStreet()).isEqualTo("street");
+        Assertions.assertThat(representativeSaved.getUserId()).isEqualTo("userId");
+    }
+
+    @Test
+    public void givenRepresentativeAndUserIdNotEquals_whenSave_thenRepresentativeThrowBadRequestDataException() {
+        // Given:
+        RepresentativeRepository representativeRepository = Mockito.mock(RepresentativeRepository.class);
+
+        RepresentativesServiceImpl representativesService = new RepresentativesServiceImpl(representativeRepository);
+
+        org.uresti.pozarreal.dto.Representative representativeDto = org.uresti.pozarreal.dto.Representative.builder()
+                .house("house")
+                .phone("123456")
+                .street("street")
+                .userId("userId")
+                .build();
+
+        // When:
+        // Then:
+        Assertions.assertThatThrownBy(() -> representativesService.save(representativeDto, "userId2"))
+                .isInstanceOf(BadRequestDataException.class)
+                .hasMessage("UserId does not match with request body to update representative", "WRONG_USER_ID");
+
+        Mockito.verifyNoMoreInteractions(representativeRepository);
+        Mockito.verifyNoInteractions(representativeRepository);
```

Comment:
It is not expected to use both sentences .. use one of them instead (I mean only verifyNoMoreInteractions or only verifyNoInteractions)